File: ios/Models/category.swift
import Foundation

struct Category: Codable, Identifiable, Hashable {
    let id: UUID
    let name: String
    let description: String?
    let isActive: Bool
    let teamId: UUID?

    enum CodingKeys: String, CodingKey {
        case id, name, description
        case isActive = "is_active"
        case teamId = "team_id"
    }
}

File: backend/app/models/category.py
import uuid

from sqlalchemy import Boolean, ForeignKey, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.session import Base


class Category(Base):
    __tablename__ = "categories"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name: Mapped[str] = mapped_column(String(100), nullable=False)
    description: Mapped[str | None] = mapped_column(Text, nullable=True)
    is_active: Mapped[bool] = mapped_column(Boolean, default=True)

    # Bu kategorideki kayitlari hangi ekip ustlenecek.
    # NULL ise kayit genel havuza duser (tum temsilciler gorur).
    team_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True), ForeignKey("teams.id"), nullable=True, index=True
    )
